src/planner/parabolic/HauserParabolicSmootherHelpers.cpp

```diff
@@ -13,172 +13,184 @@ namespace planner {
+  startTime = std::chrono::system_clock::now();
+  double elapsedTime = 0;
+
+  while (elapsedTime < timelimit && dynamicPath.ramps.size() > 3)
+  {
+    std::uniform_real_distribution<> dist(0.0, dynamicPath.GetTotalTime());
+    double t1 = dist(rng);
+    double t2 = dist(rng);
+    dynamicPath.TryShortcut(t1, t2, feasibilityChecker);
+
+    currentTime = std::chrono::system_clock::now();
+    elapsedTime = std::chrono::duration_cast<std::chrono::duration<double>>(
+                      currentTime - startTime)
+                      .count();
+  }
+  return true;
 }
 
-
-bool doBlend(ParabolicRamp::DynamicPath& dynamicPath,
-             aikido::constraint::TestablePtr testable,
-             double blendRadius, int blendIterations,
-             double checkResolution, double tolerance)
+bool doBlend(
+    ParabolicRamp::DynamicPath& dynamicPath,
+    aikido::constraint::TestablePtr testable,
+    double blendRadius,
+    int blendIterations,
+    double checkResolution,
+    double tolerance)
 {
-    if(blendIterations <= 0)
-      throw std::runtime_error("Blend iterations should be positive");
-    if(blendRadius <= 0.0)
-      throw std::runtime_error("Blend radius should be positive");
-
-    SmootherFeasibilityCheckerBase base(testable, checkResolution);
-    ParabolicRamp::RampFeasibilityChecker feasibilityChecker(&base, tolerance);
-
-    // Mark all of the ramps in the initial trajectory as "original". We'll
-    // only try to blend transitions between these ramps.
-    for(unsigned int i=0;i<dynamicPath.ramps.size();++i)
-    {
-        dynamicPath.ramps[i].blendAttempts = 0;
-    }
-
-    double dtShortcut = blendRadius;
-
-    // tryBlend always starts at the beginning of the trajectory.
-    // Without this bookkeeping, tryBlend would could any blend that fails multiple times.
-    for (int attempt = 0; attempt < blendIterations; ++attempt)
-    {
-        while (tryBlend(dynamicPath, feasibilityChecker, attempt, dtShortcut));
-
-        dtShortcut /= 2.;
-    }
-    return true;
+  if (blendIterations <= 0)
+    throw std::runtime_error("Blend iterations should be positive");
+  if (blendRadius <= 0.0)
+    throw std::runtime_error("Blend radius should be positive");
+
+  SmootherFeasibilityCheckerBase base(testable, checkResolution);
+  ParabolicRamp::RampFeasibilityChecker feasibilityChecker(&base, tolerance);
+
+  // Mark all of the ramps in the initial trajectory as "original". We'll
+  // only try to blend transitions between these ramps.
+  for (unsigned int i = 0; i < dynamicPath.ramps.size(); ++i)
+  {
+    dynamicPath.ramps[i].blendAttempts = 0;
+  }
+
+  double dtShortcut = blendRadius;
+
+  // tryBlend always starts at the beginning of the trajectory.
+  // Without this bookkeeping, tryBlend would could any blend that fails
+  // multiple times.
+  for (int attempt = 0; attempt < blendIterations; ++attempt)
+  {
+    while (tryBlend(dynamicPath, feasibilityChecker, attempt, dtShortcut))
+      ;
```

Comment:
The current logic is correct. The core issue is that blending can completely remove waypoints from the trajectory, which makes bookkeeping via the time parameter extraordinarily difficult. We dodge that issue by allowing `tryBlend` to always start at the beginning of the trajectory. We continue to call `tryBlend` in a loop until it finds nothing left to blend.